### PR TITLE
[feat]: adding endpoint forgot password

### PR DESCRIPTION
**Pull request by cristianoCatolico**
This endpoint receives an email and applicationId and send the email with the link to change password. It lefts to do the endpoint of change password. This pr also contains the definitions for endpoints of registration however I could not make it work yet.

### CODE REVIEW

File: pkg/api/api.go [hunk 1]
```diff
@@ -53,6 +53,9 @@ func (s *APIServer) Init() error {
 
 	// Auth routes
 	router.HandleFunc("POST /api/login", makeHTTPHandlerFunc(s.authHandlers.Login))
+	router.HandleFunc("POST /api/forgot-password", makeHTTPHandlerFunc(s.authHandlers.ForgotPassword))
+	router.HandleFunc("POST /api/register-user", makeHTTPHandlerFunc(s.authHandlers.RegisterUser))
```
Comment (grez-lucas) [inline pkg/api/api.go]: Regarding these endpoints, I feel we could make them more RESTful by instead using:

- `POST /api/register-user` -> `POST /api/user`
- `POST /api/verify-email` -> `POST /api/user/{id}/verify-email`

File: pkg/handlers/auth.go
Comment (grez-lucas) [inline pkg/handlers/auth.go:1]: This is on the right track! I imagine the forgotPasswordRequest is being reused for these handlers but will be switched later on

File: pkg/handlers/requests.go
Comment (grez-lucas) [inline pkg/handlers/requests.go:1]: Nice

File: pkg/api/api.go (hunk 1 above)
Comment (cristianoCatolico) [reply pkg/api/api.go]: Thanks I will change it.

File: pkg/handlers/requests.go
Comment (cristianoCatolico) [reply pkg/handlers/requests.go:1]: Thanks

File: pkg/handlers/auth.go
Comment (cristianoCatolico) [reply pkg/handlers/auth.go:1]: I dit not get the idea, but I will annotate to ask later.

File: pkg/services/auth.go [hunk 2]
```diff
@@ -364,3 +364,130 @@ func (s *AuthService) NewFusionAuthClient() (*fusionauth.FusionAuthClient, error
...
+
+	userFusion, faErr, err := client.RetrieveUser(userID)
+
+	if err != nil {
+		return nil, err
+	}
+	if faErr != nil {
+		return nil, NewFaError(userFusion.StatusCode, faErr.Error())
+	}
+	if userFusion.User.Verified {
```
Comment (grez-lucas) [inline pkg/services/auth.go:477]: Nice error handling!

**Review by grez-lucas — APPROVED** [review]
Nice PR! You're slowly becoming a GoLang expert 😎. Make sure to document these new endpoints and Request/Response objects in Apiary when you get the chance.

I expect some of the logic for the `POST /api/tenant` endpoint to require some refactors from my side after this PR, since there are new tenant and application configurations we're taking into account. 

Don't forget to commit your changes to the `kickstart.example.json` (which should be a copy of the `kickstart.json` file) **in the parent repository**. As a note here:
- I merged a PR recently where I hard-coded the Tenant1's UUID and App1 UUID, make sure your new `kickstart.example.json` includes these variable definitions ;)

**Pull request merged**